### ISSUE TITLE
[stable/prometheus] Prometheus not scraping Python Prometheus client http_server_start metrics endpoint

### ISSUE THREAD

**Issue by mdenend** [helm/charts#11715]
**Is this a request for help?**: Yes

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**: 
K8s: v1.13.3 (also have seen it on v1.10.11-gk1.1, which is where my deployment is.)
Helm: v2.12.2

**Which chart**: stable/prometheus version 8.7.1

**What happened**:  Breaking this into two different subsections:
- **What doesn't work:**
  - Prometheus doesn't appear to be scraping relabel_config endpoints started from the Python Prometheus client start_http_server(port) when adding into extraScrapeConfig scraping the app label.
  - Same as above, but running run_wsgi_server(port) instead also doesn't appear to work.

- **What does work:**
  - Returning a wrapped Flask wsgi app to gnuicorn and using the k8s app label to scrape /metrics works.
  - Changing relabel_configs to static_configs and feeding the internal Pod IP works.

**What you expected to happen**:
I expect Prometheus to be able to scrape the /metrics endpoint when start_http_server(port) is provided and the correct pod app label is provided.

**How to reproduce it** (as minimally and precisely as possible):
I've sent a zipped directory of files here: [example.zip](https://github.com/helm/charts/files/2902724/example.zip)  I've done configuration with to reproduce this bug. To actually run the test:

1. Find a k8s cluster (minikube works too.)
2. In this directory, build the Dockerfile: `docker build -f Dockerfile -t prometheus-test`
3. Run `kubectl apply -f client.yml`
4. Find the pod IP for client-prometheus: `kubectl get pods -o jsonpath='{range .items[0]}{@.metadata.labels.app}: {@.status.podIP}{"\n"}'` then replace 172.17.0.10 in prometheus-good.yml:  static_config.targets with the pod's IP.
5. Run `helm install stable/prometheus -n prometheus -f good-prometheus-override.yml`
6. Port forward Prometheus:
  - `export POD_NAME=$(kubectl get pods --namespace default -l "app=prometheus,component=server" -o jsonpath="{.items[0].metadata.name}")`
  - `kubectl --namespace default port-forward $POD_NAME 9090`
7. Observe that request_processing_seconds_count and _total can be queried. 
8. Replace good configuration with bad one: `helm upgrade prometheus stable/prometheus -f prometheus-override.yml`
9. Observe that you no longer get above counters after the config map has been updated (after 30 seconds or so.)

I can provide a simple script to also return a wrapped gunicorn Flask server and show that does work with the kubectl app labeling,  if necessary.

**Anything else we need to know**: This could be an issue with the chart or the Prometheus server itself. I wasn't sure where to open the issue, please let me know if I should send this issue to the official Prometheus server folks. 

Also, I don't think this is an issue with the Python Prometheus client, as I am able to get some scrape configurations working in some cases. But if you think this may be the case, I can open an issue with them as well.

**Comment by mdenend** [comment]
OK, it took me quite some time to figure this out, but I finally got a solution.  I created a ClusterIP by applying this yaml file to my cluster. [client-prometheus-service.txt](https://github.com/helm/charts/files/2906874/client-prometheus-service.txt) What made this especially tricky to figure out was the fact that it needed this line:

` annotations: {prometheus.io/scrape: 'true'}`

needed to to be added to the service `metadata.annotations`. 

May I suggest adding a section in the README on how to expose pods that have custom endpoints to prometheus? I feel it would save others quite some time. 

I'd be happy to take a crack at attempting to write such a section when I get the time. Also, let me know if you think I should close this issue and reopen a FEATURE REQUEST to modify the README making this part clearer.
